### PR TITLE
DD-272

### PR DESCRIPTION
**Pull request by KLV96**
Drop Node 14, 16 and 18 and allowing only 20.x || 22.x and updating the following dependencies: async ^3.2.5 -> ^3.2.6
body-parser ^1.20.2 -> ^2.0.1
debug ^4.3.4 -> ^4.3.7
helmet ^6.2.0 -> ^8.0.0
uuid ^9.0.1 -> ^10.0.0

devDependencies:
eslint ^8.57.0 -> ^9.11.1
express ^4.19.2 -> ^4.21.0
hmpo-components ^6.5.1 -> ^7.0.0
hmpo-config ^3.0.1 -> ^4.0.0
hmpo-i18n ^6.0.2 -> ^7.0.0
hmpo-logger ^7.0.2 -> ^8.0.0
husky ^8.0.3 -> ^9.1.6
mocha ^10.7.0 -> ^10.7.3
nyc ^17.0.0 -> ^17.1.0
Replacing reqres ^3.0.1 with hmpo-reqres 2.0.0
sinon ^18.0.0 -> ^19.0.2

peerDependencies:
hmpo-components >=6 -> >=7
hmpo-config >=3 -> >=4
hmpo-i18n >=6 -> >=7
hmpo-logger >=7 -> >=8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,42 +16,42 @@
...
-    "helmet": "^6.2.0",
+    "helmet": "^8.0.0",
     "nocache": "^3.0.4",
     "on-finished": "^2.4.1",
     "redis": "^4.7.0",
     "underscore": "^1.13.7",
-    "uuid": "^9.0.1"
+    "uuid": "^10.0.0"
   },
   "devDependencies": {
```
Comment (SamChatfield) [inline package.json:39]: Missing globals as a `devDependency`

File: package.json [hunk 2]
```diff
@@ -16,42 +16,42 @@
...
     "test:lint": "eslint .",
     "test:unit": "nyc _mocha",
-    "prepare": "husky install"
+    "prepare": "husky"
   },
   "dependencies": {
-    "async": "^3.2.5",
-    "body-parser": "^1.20.2",
+    "async": "^3.2.6",
+    "body-parser": "^2.0.1",
```
Comment (SamChatfield) [inline package.json:23]: This looks like an unstable version, `latest` is still `1.20.3` so we can stick with that for now. We can have another look at this when we look into updating to Express 5.

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+
+module.exports = [
+    js.configs.recommended,
+    {
+        languageOptions: {
+            ecmaVersion: 2022,
+            globals: {
+                ...globals.node,
+                APP_ROOT: 'readonly',
```
Comment (SamChatfield) [inline eslint.config.js:31]: Should be a test global

File: eslint.config.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                express: 'readonly',
+                expect: 'readonly',
+                LOGGER_RESET: 'readonly',
+                CONFIG_RESET: 'readonly',
```
Comment (SamChatfield) [inline eslint.config.js:35]: Should be test globals

File: eslint.config.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        'error',
+        { anonymous: 'always', named: 'never' },
+    ],
+    'no-mixed-spaces-and-tabs': 'error',
+    'comma-spacing': ['error', { before: false, after: true }],
+    'key-spacing': ['error', { beforeColon: false, afterColon: true }],
+};
+
+
+module.exports = [
```
Comment (SamChatfield) [inline eslint.config.js:24]: Can we add a global ignore for `example/public/**`? If you happen to have ran the example app it throws loads of lint errors on the built `application.js` without that.

File: eslint.config.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        rules: {
+            'no-unused-vars': [
+                'error',
+                { argsIgnorePattern: '^(err|req|res|next)$' },
+            ],
+            'one-var': ['error', { initialized: 'never' }],
+            'no-var': 'error',
+            'no-restricted-syntax': [
+                'off',  // Disable restrictions for certain reserved words
+            ],
```
Comment (SamChatfield) [inline eslint.config.js:47]: Should be able to remove this, it doesn't actually do anything with this config.

File: eslint.config.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+module.exports = [
+    js.configs.recommended,
+    {
+        languageOptions: {
+            ecmaVersion: 2022,
+            globals: {
+                ...globals.node,
+                APP_ROOT: 'readonly',
+                express: 'readonly',
```
Comment (SamChatfield) [inline eslint.config.js:32]: Not an actual global used anywhere in here, can remove this.

File: middleware/index.js [hunk 8]
```diff
@@ -1,3 +1,5 @@
+/* istanbul ignore file */
```
Comment (SamChatfield) [inline middleware/index.js:1]: We don't want to ignore coverage here

**Review by SamChatfield — CHANGES_REQUESTED** [review]
A few changes needed.

We also need to remove `test/.eslintrc`

**Review by msjhall138 — APPROVED** [review]

**Review by imran-gov — APPROVED** [review]

File: eslint.config.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    ],
+    'no-mixed-spaces-and-tabs': 'error',
+    'comma-spacing': ['error', { before: false, after: true }],
+    'key-spacing': ['error', { beforeColon: false, afterColon: true }],
+};
+
+module.exports = [
+    {
+        ignores: ['example/public/**'],
+        ...js.configs.recommended,
```
Comment (SamChatfield) [inline eslint.config.js:25]: This isn't the right place for the base config to go

**Review by SamChatfield — CHANGES_REQUESTED** [review]

File: eslint.config.js (hunk 5 above)
Comment (SamChatfield) [reply eslint.config.js:22]: The [global ignore](https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores) for this hasn't gone in the right place

File: eslint.config.js (hunk 9 above)
Comment (SamChatfield) [reply eslint.config.js:25]: See internal ESLint config for how to do a base and global ignores

**Review by SamChatfield — APPROVED** [review]
LGTM

**Pull request merged**